qiita_db/artifact.py

```diff
@@ -420,6 +421,15 @@ def _associate_with_analysis(instance, analysis_id):
                 sql_args = [(instance.id, p.id) for p in parents]
                 qdb.sql_connection.TRN.add(sql, sql_args, many=True)
 
+                # inheriting visibility
```

Comment:
@antgonza can you review this section? This was added in master and I had a bit of trouble solving the conflicts in here but I think I end up placing this code in the correct location.